### PR TITLE
Using an adjustable wire time window for CSC RecHit building

### PR DESCRIPTION
**Pull request by mhl0116**
Use a configurable wire time window for CSC RecHit reconstruction.

Current wire time window is from 0-15, the goal is to make this window adjustable(to a narrower one, e.g from 6-10).
The study has been done to show:
1, narrow wire time window reduces fake RecHit/Segment at high luminosity condition
2, preserves muon segment reconstruction efficiency

More details are in this twiki: https://twiki.cern.ch/twiki/bin/viewauth/CMS/UFCSCLocalRecoStudy

### CODE REVIEW

**Comment by slava77** [comment]
@ptcox please confirm the changes are needed and were approved by the DPG.

@mhl0116 please edit the PR title (there is an "Edit" button to the right from the title) so that it is self-documenting and descriptive (think of a paper/article title).
Also, please send an email to me to clarify your identity (or, if appropriate, update your github account to show your real name).

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by ptcox** [comment]
Hi Slava,

I presume this is from Hualin Mei. a U Florida grad student who has been 
working on improving the CSC rechit building. This is an implementation 
of their well documented studies of improvements from reducing time 
windows for association of wires and strips in rechit building.
I haven't checked the code myself yet, but we asked that it be 
configurable, and OFF by default i.e. no change from existing code.

Regards, Tim

Slava Krutelyov wrote:

**Comment by slava77** [comment]
```
----- Begin Fatal Exception 15-Aug-2016 18:02:41 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=CSCRecHitDProducer label='csc2DRecHits'
Exception Message:
ValueError type of CSCUseReducedWireTimeWindow is expected to be int but declared as bool
----- End Fatal Exception -------------------------------------------------

```
the problem is related to the changes made

Please test the code before submitting
http://cms-sw.github.io/PRWorkflow
the successful run of ``` runTheMatrix.py -l limited -i all ``` is required before submitting a PR.

**Comment by slava77** [comment]
IN 136.731 (singlePhoton 2016B workflow)
all_oldvsnew_runsingleph2016bwf136p731c_hbherechitssorted_hbhereco__rereco_obj_obj_time
is this expected?

**Comment by slava77** [comment]
sorry, last comment about HBHE timing was posted in wrong PR

**Comment by mhl0116** [comment]
updated HLTrigger/Configuration to fix previous problem during test.
Perfom runTheMatrix.py -l limited -i all, only 1330 failed, the reason is cannot access to this sample:
file dataset=/RelValZMM_13/CMSSW_7_6_0_pre7-76X_mcRun2_asymptotic_v9_realBS-v1/GEN-SIM site=T2_CH_CERN
I looked up to DAS, this sample is no longer accessible to user, any help or work around is appreciated

logs are here: http://tier2.ihepa.ufl.edu/~mhl/2016/pullRequest_15470/

**Comment by Martin-Grunewald** [comment]
-1
Changes to HLT menus in HLTrigger/Configuration have to go through ConfDB (if this is also in 80X)
or a special customisation in HLTrigger/Configuration/python/customizeHLTforCMSSW.py
(The large dumps are NEVER modified directly!)

**Comment by slava77** [comment]
::fillDescriptions method should help, if I'm not mistaken.
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp

**Comment by mhl0116** [comment]
use fillDescription to add new configurations for adjustable wire time window in RecoLocalMuon/CSCRecHitD/src/CSCRecHitDProducer.cc

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoLocalMuon/CSCRecHitD/src/CSCHitFromWireOnly.cc [hunk 1]
```diff
@@ -76,7 +80,17 @@ std::vector<CSCWireHit> CSCHitFromWireOnly::runWire( const CSCDetId& id, const C
...
+
+      if (!useReducedWireTime) {
+
+         hitsInLayer.push_back( whit );
+
+         } else if (theTime >= wireTimeWindow_low && theTime <= wireTimeWindow_high) {
+
+                   hitsInLayer.push_back( whit );	
+
+                   }
```
Comment (slava77) [inline RecoLocalMuon/CSCRecHitD/src/CSCHitFromWireOnly.cc]: please use consistent indentation.
also, most of the empty lines are not needed in the added code

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/15470 	ed022db
- changes are in line with the PR description and the follow up review. The default behavior is unchanged.
- jenkins tests pass and comparisons with baseline show no difference.

**Comment by ptcox** [comment]
Why does this need discussion in the ORP meeting?

cmsbuild wrote:

**Pull request merged**